boot/bootutil/src/loader.c

```diff
@@ -1998,6 +2002,10 @@ context_boot_go(struct boot_loader_state *state, struct boot_rsp *rsp)
     }
 #endif
 
+#if defined(CONFIG_MCUBOOT_ACTION_HOOKS)
```

Comment:
I think it would be a lot cleaner if, instead of an ifdef around every call of the status change, we made the function always defined, but when the feature is not defined, make it an inline function with an empty body (and the incantation to ignore the argument).